mitodl/odl-video-service
# TechTV import

#### What are the relevant tickets?
- Closes #414
- Closes #415
- Closes #417 
- Closes #428

#### What's this PR do?
- Migrates TechTV collections, videos, thumbnails, MIT users to OVS.

#### How should this be manually tested?
- Install/start MySQL and import the TechTV database dump located at https://www.dropbox.com/home/ODL-Video/TechTV/SQL
   ```
    mysql -u <username> -p techtv <dump_file.sql>
    ```

- Run the django command:
  ```
  docker-compose run web python manage.py techtvimport --user <mysql_user>  --db techtv --host 10.0.2.2 --aws --collection 3033
  ```
- This should import one protected collection and all its videos.  
- After the celery tasks to copy over S3 video files are finished, the status of the imported videos should change to 'Complete' .
- Videos should be playable.  
- There will not be any captions to import.
- Thumbnails should be visible on the collection page. 
- The collection owner should be the same as in TechTV if the email address ends with `@mit.edu`, otherwise it should be the same as `settings.LECTURE_CAPTURE_USER` 
- You can run the following command to test processing all collections & videos but skip copying over the S3 files. Videos and collections will be created in OVS, but thumbnails won't display and videos won't play:
  ```
  docker-compose run web python manage.py techtvimport --user <mysql_user>  --db techtv --host 10.0.2.2
  ```
